src/Microsoft.DocAsCode.Build.Engine/DocumentBuilder.cs

```diff
@@ -107,6 +107,16 @@ public void Build(DocumentBuildParameters parameters)
                         UpdateHref(manifest, context);
 
                         var generatedManifest = TemplateProcessor.Transform(processor, manifest.Select(s => s.Item).ToList(), context, parameters.ApplyTemplateSettings);
+
+                        // export sitemap
+                        YamlUtility.Serialize(
+                            Path.Combine(parameters.OutputBaseDir, "site.xref"),
+                            from xref in context.XRefSpecMap.Values
+                            select new XRefSpec(xref)
+                            {
+                                Href = ((RelativePath)context.FileMap[xref.Href]).RemoveWorkingFolder().ToString() + "#" + XrefDetails.GetHtmlId(xref.Uid),
+                            });
```

Comment:
extract to a data model? And from the experience of Manifest file/TOC, a object model is more extensible than an array model....